src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultTableRenderer.cs

```diff
@@ -44,42 +32,8 @@ public void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> comple
             _loggerWithColor.LogMinimal(SourceSeparator);
             _loggerWithColor.LogMinimal($"Source: {source.Name} ({source.SourceUri})");
 
-            Table table;
-
-            if (_verbosity == PackageSearchVerbosity.Minimal)
-            {
-                if (_exactMatch)
-                {
-                    table = new Table(_minimalColumnsToHighlight, _minimalVerbosityTableHeaderForExactMatch);
-                }
-                else
-                {
-                    table = new Table(_minimalColumnsToHighlight, _minimalVerbosityTableHeader);
-                }
-            }
-            else if (_verbosity == PackageSearchVerbosity.Detailed)
-            {
-                if (_exactMatch)
-                {
-                    table = new Table(_detailedColumnsToHighlight, _detailedVerbosityTableHeaderForExactMatch);
-                }
-                else
-                {
-                    table = new Table(_detailedColumnsToHighlight, _detailedVerbosityTableHeader);
-                }
-            }
-            else
-            {
-                if (_exactMatch)
-                {
-                    table = new Table(_normalColumnsToHighlight, _normalVerbosityTableHeaderForExactMatch);
-                }
-                else
-                {
-                    table = new Table(_normalColumnsToHighlight, _normalVerbosityTableHeader);
-                }
-            }
-
+            ITableFormatStrategy strategy = TableFormatStrategyFactory.GetStrategy(_verbosity);
+            Table table = strategy.CreateTable(completedSearch, _exactMatch);
```

Comment:
If exact match changes the columns, isn't it a strategy itself, rather than something that's part of the table creation?